catalog/security/banner-marking/src/main/java/org/codice/alliance/security/banner/marking/BannerCommonMarkingExtractor.java

```diff
@@ -0,0 +1,175 @@
+        tempMap.put(SECURITY_CLASSIFICATION_SYSTEM, this::processClassSystem);
+
+        setAttProcessors(tempMap);
+    }
+
+    Attribute processClassMarking(Metacard metacard, BannerMarkings bannerMarkings) {
+        Attribute currAttr = metacard.getAttribute(SECURITY_CLASSIFICATION);
+        if (currAttr != null) {
+            BannerMarkings.ClassificationLevel currClassLevel =
+                    BannerMarkings.ClassificationLevel.lookupByShortname(
+                            (String) currAttr.getValue());
+
+            BannerMarkings.ClassificationLevel higherLevel =
+                    BannerMarkings.ClassificationLevel.getHigherLevel(currClassLevel,
+                            bannerMarkings.getClassification());
+
+            return new AttributeImpl(SECURITY_CLASSIFICATION,
+                    ImmutableList.<String>of(higherLevel.getShortName()));
+        } else {
+            return new AttributeImpl(SECURITY_CLASSIFICATION, ImmutableList.<String>of(
+                    bannerMarkings.getClassification()
+                            .getShortName()));
+        }
+    }
+
+    Attribute processReleasability(Metacard metacard, BannerMarkings bannerMarkings) {
+        Attribute currAttr = metacard.getAttribute(SECURITY_RELEASABILITY);
+        if (currAttr != null) {
+            return new AttributeImpl(SECURITY_RELEASABILITY,
+                    dedupedList(bannerMarkings.getRelTo(), currAttr.getValues()));
+        } else {
+            return new AttributeImpl(SECURITY_RELEASABILITY,
+                    ImmutableList.<String>copyOf(bannerMarkings.getRelTo()));
+        }
+    }
+
+    Attribute processCodewords(Metacard metacard, BannerMarkings bannerMarkings) {
+        List<String> sciControls = new ArrayList<>();
+        for (BannerMarkings.SciControl sci : bannerMarkings.getSciControls()) {
+            if (sci.getCompartments()
+                    .isEmpty()) {
+                sciControls.add(sci.getControl());
+                continue;
+            }
+            for (Map.Entry<String, List<String>> comp : sci.getCompartments()
+                    .entrySet()) {
+                StringBuilder sb = new StringBuilder(sci.getControl());
+                sb.append("-");
+                sb.append(comp.getKey());
+                if (comp.getValue()
+                        .isEmpty()) {
+                    sciControls.add(sb.toString());
+                    continue;
+                }
+                sb.append(comp.getValue()
+                        .stream()
+                        .collect(Collectors.joining(" ", " ", "")));
+                sciControls.add(sb.toString());
+            }
+        }
+
+        // Handle NATO markings if present
+        if (bannerMarkings.getType() == FGI && !StringUtils.isEmpty(
+                bannerMarkings.getNatoQualifier())) {
+            sciControls.add(bannerMarkings.getNatoQualifier());
```

Comment:
In glancing at the [IC-ISM-v2.1.xsd schema](http://schemas.opengis.net/ic/2.1/IC-ISM-v2.1.xsd), I'm wondering if I should handle NATO qualifiers differently and if I should mark NATO/COSMIC classified documents differently. 

E.g. if I follow the `ClassificationType` enumeration in that schema, instead of marking a `//COSMIC TOP SECRET//BOHEMIA` document with classification of `TS` and an added SCI control of `BOHEMIA`, perhaps I should mark it with a classification of `CTS-B`.

@stustison what are your thoughts on this?